creator-node/src/services/stateMachineManager/stateMonitoring/stateMonitoringUtils.js

```diff
@@ -0,0 +1,409 @@
+
+        // Error case 2 - already marked unhealthy
+      } else if (unhealthyPeers.has(secondary)) {
+        logger.error(
+          `processStateMachineOperation Secondary ${secondary} for user ${wallet} in unhealthy peer set. Marking replica as unhealthy.`
+        )
+        unhealthyReplicas.add(secondary)
+
+        // Error case 3 - low user sync success rate
+      } else if (
+        failureCount >= MIN_FAILED_SYNC_REQUESTS_BEFORE_RECONFIG &&
+        successRate < MIN_SECONDARY_USER_SYNC_SUCCESS_PERCENT
+      ) {
+        logger.error(
+          `processStateMachineOperation Secondary ${secondary} for user ${wallet} has userSyncSuccessRate of ${successRate}, which is below threshold of ${MIN_SECONDARY_USER_SYNC_SUCCESS_PERCENT}. ${successCount} Successful syncs vs ${failureCount} Failed syncs. Marking replica as unhealthy.`
+        )
+        unhealthyReplicas.add(secondary)
+
+        // Success case
+      } else {
+        potentialSyncRequests.push({ ...nodeUser, endpoint: secondary })
+      }
+    }
+
+    /**
+     * If any unhealthy replicas found for user, enqueue an updateReplicaSetOp for later processing
+     */
+    if (unhealthyReplicas.size > 0) {
+      requiredUpdateReplicaSetOps.push({ ...nodeUser, unhealthyReplicas })
+    }
+
+    /**
+     * If this node is secondary for user, check both secondaries for health and enqueue SyncRequests against healthy secondaries
+     * Ignore unhealthy secondaries for now
+     */
+  } else {
+    // filter out false-y values to account for incomplete replica sets and filter out the
+    // the self node
+    replicaSetNodesToObserve = [
+      { endpoint: primary, spId: primarySpID },
+      ...replicaSetNodesToObserve
+    ]
+    replicaSetNodesToObserve = replicaSetNodesToObserve.filter((entry) => {
+      return entry.endpoint && entry.endpoint !== thisContentNodeEndpoint
+    })
+
+    for (const replica of replicaSetNodesToObserve) {
+      // If the map's spId does not match the query's spId, then regardless
+      // of the relationship of the node to the user, issue a reconfig for that node
+      if (endpointToSPIdMap[replica.endpoint] !== replica.spId) {
+        unhealthyReplicas.add(replica.endpoint)
+      } else if (unhealthyPeers.has(replica.endpoint)) {
+        // Else, continue with conducting extra health check if the current observed node is a primary, and
+        // add to `unhealthyReplicas` if observed node is a secondary
+        let addToUnhealthyReplicas = true
+
+        if (replica.endpoint === primary) {
+          addToUnhealthyReplicas = !(await isPrimaryHealthy(primary))
+        }
+
+        if (addToUnhealthyReplicas) {
+          unhealthyReplicas.add(replica.endpoint)
+        }
+      }
+    }
+
+    if (unhealthyReplicas.size > 0) {
+      requiredUpdateReplicaSetOps.push({ ...nodeUser, unhealthyReplicas })
+    }
+  }
+
+  return { requiredUpdateReplicaSetOps, potentialSyncRequests }
+}
+
+module.exports = {
```

Comment:
I'm open to talking about this and standardizing. I like defining `module.exports` and the bottom because it makes it easy to scroll to the bottom of any file and immediately know which functions are exposed

Reply:
good with me, not sure which way i like better, nbd anyway